runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -298,6 +301,18 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
       {
       vmemParams.pageSize = largeCodePageSize;
       vmemParams.pageFlags = config.largeCodePageFlags();
+
+      // Recalculate the page size if using large pages
+      if(vmemParams.pageSize > ccTotalSizeB)
+         {
+         for (UDATA pageIndex = 0; 0 != pageSizes[pageIndex]; ++pageIndex)
+            {
+            if(pageSizes[pageIndex] <= ccTotalSizeB)
+               vmemParams.pageSize = pageSizes[pageIndex];
+            }
+         if (config.verboseCodeCache())
+            TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE,"Warning:Using page size %d instead of large page size %d", vmemParams.pageSize, largeCodePageSize);
```

Comment:
`vmemParams.pageSize;` is uintptr_t, so %d is not correct here. `largeCodePageSize` may have the same problem